Interview-Exercise/PersistRepository.cs

```diff
@@ -0,0 +1,143 @@
+                foreach (string file in files)
+                    File.Delete(file);
+            }
+        }
+
+        void IPersistRepository.Delete(string countryCode)
+        {
+            if (IsBaseDirectoryThere())
+            {
+                string filePath = BuildFullPath(countryCode);
+                if (File.Exists(filePath))
+                {
+                    string fileData = File.ReadAllText(filePath);
+                    int startIndex = fileData.IndexOf(countryCode);
+                    if (startIndex != -1)
+                    {
+                        string newFileContent = fileData.Remove(startIndex, startIndex + 8); // 8 = 3 letters plus 1 comma multiplied by 2 
+                        File.WriteAllText(filePath, newFileContent);
+                    }
+                    else throw new CountryNotFoundException(countryCode);
+                }
+                else throw new CountryNotFoundException(countryCode);
+            }
+        }
+
+        Country IPersistRepository.Get(string countryCode)
+        {
+            if (IsBaseDirectoryThere())
+            {
+                string filePath = BuildFullPath(countryCode);
+                if (File.Exists(filePath))
+                {
+                    string fileData = File.ReadAllText(filePath);
+                    int startIndex = fileData.IndexOf(countryCode);
+                    if (startIndex != -1)
+                    {
+                        string entry = fileData.Substring(startIndex, startIndex + 8); // 8 = 3 letters plus 1 comma multiplied by 2 
+                        string[] tokens = entry.Split(new char[] { ',' });
+                        return new Country(tokens[0], tokens[1]);
+                    }
+                    else throw new CountryNotFoundException(countryCode);
+                }
+                else throw new CountryNotFoundException(countryCode);
+            }
+
+            throw new CountryNotFoundException();
+        }
+
+        void IPersistRepository.Update(Country country)
+        {
+            if (IsBaseDirectoryThere())
+            {
+                string filePath = BuildFullPath(country);
+                if (File.Exists(filePath))
+                {
+                    string fileData = File.ReadAllText(filePath);
+                    int startIndex = fileData.IndexOf(country.Code);
+                    if (startIndex != -1)
+                    {
+                        fileData = fileData.Remove(startIndex, startIndex + 8); // 8 = 3 letters plus 1 comma multiplied by 2 
+                        fileData = fileData.Insert(startIndex, BuildCountryEntry(country));
+                        File.WriteAllText(filePath, fileData);
+                    }
+                    else throw new CountryNotFoundException(country.Code);
+                }
+                else throw new CountryNotFoundException(country.Code);
+            }
+        }
```

Comment:
Should throw Country not found exception in the case the base directory is not there.